include/cubos/settings.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef CUBOS_SETTINGS_HPP
+#define CUBOS_SETTINGS_HPP
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+namespace cubos
+{
+
+    class Settings
+    {
+    private:
+        Settings() = default;
+        static Settings* _instance;

```

Comment:
Why aren't this methods and fields in line 36? Also, are we using underscore in private fields? I wasn't following that convention in the rest of the code, since I personally don't like doing that in C++: names starting with _ in C++ are reserved.

I also think we should use std::unique_ptr<Settings> instead of a raw pointer, since the current code leads to a memory leak.

Reply:
Regarding the pointer issue,
If we use a unique_ptr, then I would not be able to return it from GetInstance. Or am I missing something?

To avoid memory leaks wouldn't it be enough to define the destructor of Settings?